### DISCUSSION TITLE
DatabaseView join query field naming convention

### DISCUSSION THREAD

**Discussion by furatamasensei** [vitusortner/floor#688] [Q&A] [answered]
Extending this closed issue about the join query (#213). Please let me know if I should ask in there instead.

I have question regarding the field / property naming convention. Let's say I set my table name on my entities to be `categories` and `menus`. If I want to join them in a view and take the `id` of each, should I name my property `categoriesId` and `menusId`?

Or do I need to alias them first? And what about the other properties?

```dart
import 'package:floor/floor.dart';

@DatabaseView(
  'SELECT'
  ' categories.id, categories.name, categories.label_color,'
  ' categories.created_at, categories.updated_at, menus.id,'
  ' menus.name, menus.price, menus.created_at, menus.updated_at '
  'FROM menus'
  'INNER JOIN categories ON categories.id = menus.category_id',
)
class CategoryMenu {
  late int menusId; // Should this be [menuId] or [menusId] ?
  late int categoriesId; // Should this be [categoryId] or [categoriesId] ?

  // Should the rest be like this?
  late String menusName;
  late String categoriesName;

  // Or like this?
  late String menuName;
  late String categoryName;

  // if I want to use [menuId] instead of [menusId], should I make an alias on
  // my query for each field?
  // Like.. "SELECT menus.id AS menuId" ?
}
```

**Comment by dkaera** [comment, marked as answer]
@pratamatama I think we should follow the Dart language naming convention anyway. This means [avoid using the possessive ](https://docs.flutter.dev/brand#:~:text=or%20verb%2C%20and-,never%20in%20the%20plural%20or%20possessive%20form.,-Use%20a%20generic)forms in naming.
So in your case will be correct:

```dart
import 'package:floor/floor.dart';

@DatabaseView(
  'SELECT'
  ' categories.id, categories.name, categories.label_color,'
  ' categories.created_at, categories.updated_at, menus.id,'
  ' menus.name, menus.price, menus.created_at, menus.updated_at '
  'FROM menus'
  'INNER JOIN categories ON categories.id = menus.category_id',
)
class CategoryMenu {
  late int menuId;
  late int categoryId;

  late String menuName;
  late String categoryName;
}
```

>   // if I want to use [menuId] instead of [menusId], should I make an alias on
  // my query for each field?
  // Like.. "SELECT menus.id AS menuId" ?
  
I think using aliases is better if some expression occurs multiple times. Don't see any reason to use it here 🤷‍♂️

**Reply by furatamasensei** [reply]
@dkaera So that mean I should use the singular form instead of the plural form for table names? It's kinda weird for me since I used to write table names plural when working with databases. But eh, I'll just need to adapt myself to it. 😄

Thank you very much!
